R/dplyr.R

```diff
@@ -25,3 +26,47 @@ mutate.skim_df <- function(.data, ...) {
   mutated <- NextMethod("mutate")
   rebuild_skim_obj(mutated, .data)
 }
+
+#'  Transmute a skim_df
+#'  
+#' [dplyr::transmute()] older than version 0.8.0 drops attributes, but we need to keep 
+#' them around for other skim behaviors and consistency with newer versions of dplyr. 
+#' Otherwise the behavior is exactly the same as dplyr 0.8.0. For more
+#' information, see https://github.com/tidyverse/dplyr/issues/3429. We also need to
+#' retain the two metadata columns (`variable` and `type`) that are the defining 
+#' features of a skim_df object.
+#'
+#' @param .data A `skim_df`, which behaves like a `tbl.`
+#' @param ... Name-value pairs of expressions, each with length 1 or the same
+#'   length as the number of rows in the group (if using [group_by()] or in the
+#'   entire input (if not using groups). The name of each argument will be the
+#'   name of a new variable, and the value will be its corresponding value. 
+#'   Use `NULL` value in `mutate` to drop a variable.  New variables overwrite
+#'   existing variables of the same name.
+#'   @return A `skim_df` object, which also inherits the class(es) of the input
+#'   data. The `variable`` and `type` variables will be included
+#'   even if not named. 
+#'   In many ways, the object behaves like a [tibble::tibble()].
+#' @seealso [dplyr::transmute()] for the function's expected behavior.
+#' @export
+transmute.skim_df <- function(.data, ...) {
```

Comment:
I don't like changing the way transmute behaves within our package. My impression is that users will be much more familiar with transmute in dplyr than in skim.

Instead, we could warn the user that they are dropping relevant skim attributes and return a tibble always.

It would be nice to have a check whether an object `is_skim_df`:

```
is_skim_df <- function(object) {
  # has our class
  # has our columns
  # has our attributes
}

```

Reply:
I'll handle that in a separate branch